src/coreclr/tests/src/Interop/UnmanagedCallersOnly/UnmanagedCallersOnlyTest.cs

```diff
@@ -759,6 +767,15 @@ .locals init (native int V_0)
 
         IntNativeMethodInvoker testNativeMethod = (IntNativeMethodInvoker)testUnmanagedCallersOnly.CreateDelegate(typeof(IntNativeMethodInvoker));
 
-        Assert.Throws<NotSupportedException>(() => testNativeMethod());
+        if (RuntimeInformation.ProcessArchitecture == Architecture.X86)
+        {
+            // This is due to a semantic change in NDirect::MarshalingRequired().
+            // See https://github.com/dotnet/runtime/issues/38697#issuecomment-653310319.
```

Comment:
May we can add `bool bUnmanagedCallersOnly` argument for `NDirect::MarshalingRequired` and alter the exact set of checks it does in that case?